ykrt/src/trace/mod.rs

```diff
@@ -57,7 +57,13 @@ pub(crate) trait TraceRecorder {
     fn stop(self: Box<Self>) -> Result<Box<dyn AOTTraceIterator>, InvalidTraceError>;
 }
 
-/// An iterator which [TraceRecord]s use to process a trace into [TraceAction]s.
+/// An iterator which [TraceRecord]s use to process a trace into [TraceAction]s. The iterator must
+/// respect the following:
+///
+/// 1. The control point call must not appear at the head of the trace.
```

Comment:
I think I know what you mean, but it's worth clarifying.

The first item handed out by the iterator should be the (sole?) immediate successor of the block containing the control point call that caused the trace to be recorded?

Reply:
If I could parse that sentence, I might agree :p

Reply:
I'm just adding more detail, as this is very nuanced code and we are likely going to have to revisit these comments later, for ourselves and others. The facts I know are:

 - The mappable block (call it `Bc`) containing the call-site of the control point doesn't appear in the trace.
 - The unmappable body of the control point is removed.
 - The first block given out is the successor block of `Bc`.

Reply:
Fixed in d5b916e.